### PR TITLE
Fix issues in PV, PVC and Mount scenarios

### PR DESCRIPTION
**Pull request by CindyXing**
In K8s, it is required that mountpoint name needs to be the same as the volume name. 
From OS perspective, a module can mount to a volume with a single mount point

In K8s, PVC name can be different from PV name.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs [hunk 1]
```diff
@@ -322,12 +323,18 @@ V1Volume GetVolume(KubernetesModule module, Mount mount)
             // Volume name will be customer defined name or modulename + mount.source
             if (this.persistentVolumeName.HasValue)
             {
-                volumeName = this.persistentVolumeName.OrDefault();
+                var volumeNameTmp = this.persistentVolumeName.OrDefault();
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs]:
```suggestion
                var pvName = this.persistentVolumeName.OrDefault();
```

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs [hunk 2]
```diff
@@ -322,12 +323,18 @@ V1Volume GetVolume(KubernetesModule module, Mount mount)
             // Volume name will be customer defined name or modulename + mount.source
             if (this.persistentVolumeName.HasValue)
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs]: if-else blocks is very confusing here. I suggest to make it a bit more easy to grasp:
```csharp
if (this.persistentVolumeName.HasValue)
{
    ...
   return new V1Volume(...);
}

if (this.storageClassName.HasValue)
{
    return new V1Volume(...);
}

return V1Volume(...);
```

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs [hunk 3]
```diff
@@ -56,18 +58,23 @@ V1PersistentVolumeClaim ExtractPvc(KubernetesModule module, Mount mount, IDictio
...
-                persistentVolumeClaimSpec.VolumeName = this.persistentVolumeName.OrDefault();
+                string volumeNameTmp = this.persistentVolumeName.OrDefault();
+                if (volumeNameTmp != volumeName)
+                {
+                    throw new InvalidModuleException(string.Format("The mount name {0} has to be the same as the PV name {1}", volumeName, volumeNameTmp));
+                }
             }
-            else if (this.storageClassName.HasValue)
+
+            persistentVolumeClaimSpec.VolumeName = volumeName;
```
Comment (dmolokanov) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs]: Is it expected to change here? it used to ignore given VolumeName when `persistentVolumeName=Some(name)`. Not sure I understood changed logic here.

Comment (darobs) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs]: This works fine for when PersistentVolume is specified, but this isn't working for the storage class case - PVC is still pending.

Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs]: OK

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs [hunk 4]
```diff
@@ -139,8 +140,7 @@ public void EmptyDirMappingForVolume()
             var config = new KubernetesConfig("image", CreatePodParameters.Create(labels: labels, hostConfig: hostConfig), Option.None<AuthConfig>());
             var docker = new DockerModule("module1", "v1", ModuleStatus.Running, RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
             var module = new KubernetesModule(docker, config);
-            var mapper = CreateMapper(storageClassName: null);
-
+            var mapper = new KubernetesDeploymentMapper("namespace", "edgehub", "proxy", Option.None<string>(), "configPath", "configVolumeName", "configMapName", "trustBundlePAth", "trustBundleVolumeName", "trustBundleConfigMapName", string.Empty, null, "apiVersion", new Uri("http://workload"), new Uri("http://management"));
```
Comment (dmolokanov) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs]: I think it's better to user CreateMapper method. If its signature lacks of some parameters lets add it

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs [hunk 5]
```diff
@@ -186,13 +201,13 @@ public void PvcMappingForStorageClassVolume()
             var config = new KubernetesConfig("image", CreatePodParameters.Create(labels: labels, hostConfig: hostConfig), Option.None<AuthConfig>());
             var docker = new DockerModule("module1", "v1", ModuleStatus.Running, RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
             var module = new KubernetesModule(docker, config);
-            var mapper = CreateMapper(storageClassName: "azureFile");
+            var mapper = CreateMapper(string.Empty, "azureFile");
```
Comment (dmolokanov) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs]:
```suggestion
            var mapper = CreateMapper(storageClassName: "azureFile");
```

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs [hunk 6]
```diff
@@ -161,13 +175,14 @@ public void PvcMappingForVolumeNameVolume()
             var config = new KubernetesConfig("image", CreatePodParameters.Create(labels: labels, hostConfig: hostConfig), Option.None<AuthConfig>());
             var docker = new DockerModule("module1", "v1", ModuleStatus.Running, RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
             var module = new KubernetesModule(docker, config);
-            var mapper = CreateMapper(persistentVolumeName: "elephant");
+            var mapper = CreateMapper("a-volume", null);
```
Comment (dmolokanov) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs]:
```suggestion
            CreateMapper(persistentVolumeName: "a-volume");
```

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs [hunk 7]
```diff
@@ -210,13 +225,13 @@ public void PvcMappingForDefaultStorageClassVolume()
             var config = new KubernetesConfig("image", CreatePodParameters.Create(labels: labels, hostConfig: hostConfig), Option.None<AuthConfig>());
             var docker = new DockerModule("module1", "v1", ModuleStatus.Running, RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
             var module = new KubernetesModule(docker, config);
-            var mapper = CreateMapper();
+            var mapper = CreateMapper(string.Empty, string.Empty);
```
Comment (dmolokanov) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs]:
```suggestion
            var mapper = CreateMapper();
```

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs [hunk 8]
```diff
@@ -225,6 +240,57 @@ public void PvcMappingForDefaultStorageClassVolume()
...
+
+            var volumes = new[]
+            {
+                parameters.Volumes.OrDefault().Single(),
+            };
+
+            var config = new KubernetesConfig("image", CreatePodParameters.Create(volumes: volumes, hostConfig: hostConfig), Option.None<AuthConfig>());
+            var docker = new DockerModule("module1", "v1", ModuleStatus.Running, RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
+            var module = new KubernetesModule(docker, config);
+            var mapper = CreateMapper(string.Empty, string.Empty);
```
Comment (dmolokanov) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesDeploymentMapperTest.cs]:
```suggestion
            var mapper = CreateMapper();
```

**Pull request merged**